pkg/kubelet/prober/prober.go

```diff
@@ -231,11 +231,7 @@ func (p *prober) newExecInContainer(container api.Container, containerID kubecon
 		var buffer bytes.Buffer
 		output := ioutils.WriteCloserWrapper(&buffer)
 		err := p.runner.ExecInContainer(containerID, cmd, nil, output, output, false, nil)
-		if err != nil {
-			return nil, err
-		}
-
-		return buffer.Bytes(), nil
+		return buffer.Bytes(), err
```

Comment:
For a clean signature, I had in mind:

```
type Cmd interface {
  CombinedOutput() ([]byte, int, error)
  Output() ([]byte, int, error)
  ...
}
```

I haven't checked other consequences of this.

A short-term compromise would be to add godocs to the `Cmd` interface (maybe in the implementation as well) describing that the bytes array has contents in case of ExitErrors.